presto-iceberg/src/test/java/com/facebook/presto/iceberg/TestIcebergMetadataListing.java

```diff
@@ -137,7 +148,9 @@ public void testTableDropWithMissingMetadata()
         tableMetadataDir.delete();
 
         assertQueryFails("SELECT * FROM iceberg.test_metadata_schema.iceberg_table1", "Table metadata is missing");
-        assertQuerySucceeds("DROP TABLE iceberg.test_metadata_schema.iceberg_table1");
+
+        long expectedDropTableDuration = Long.parseLong(testIcebergProperties.get("iceberg.hive.table-refresh.max-retry-time").replaceAll("[^0-9]", ""));
+        assertQuerySucceedsWithTimeout("DROP TABLE iceberg.test_metadata_schema.iceberg_table1", expectedDropTableDuration);
```

Comment:
Can you separate this into a different test case, and use `@Test(timeOut = ...)` instead of measuring manually?